src/protocol/communication.rs

```diff
@@ -71,21 +71,19 @@ pub fn send(stream: &mut TcpStream, message: &serde_json::Value) -> BoxResult<()
             }
         }
     }
-    Err(Box::new(simple_error::simple_error!(
-        "timed out while attempting to send status-message to {}",
-        stream.peer_addr()?
-    )))
+    let err = simple_error::simple_error!("timed out while attempting to send status-message to {}", stream.peer_addr()?);
+    Err(Box::new(err))
 }
 
 /// receives the length-count of a pending message over a client-server communications stream
-fn receive_length(stream: &mut TcpStream, alive_check: fn() -> bool, results_handler: &mut dyn FnMut() -> BoxResult<()>) -> BoxResult<u16> {
+fn receive_length(stream: &mut TcpStream, alive_check: fn() -> bool, handler: &mut dyn FnMut() -> BoxResult<()>) -> BoxResult<usize> {
```

Comment:
Should this really be `usize`?

The protocol itself encodes the length of a message into two bytes, so `u16` covers all possibilities. I know that for *most* processor architectures, optimal performance is achieved when working at a native wordsize, but the performance cost of truncating a value here is negligible (it gets checked once, then it's discarded, and it's not like any mainstream 64-bit CPU doesn't have 16-bit register operations for comparison).

However, a bigger problem exists when working with 8-bit (or smaller) architectures, for which rperf has known use-cases. `usize` will be a single byte on those platforms, resulting in truncation of the full 16-bit value.